cdm/core/src/main/java/ucar/nc2/dataset/VariableDS.java

```diff
@@ -609,172 +612,211 @@ public void removeCoordinateSystem(CoordinateSystem cs) {
-    return scaleMissingUnsignedProxy.isFillValue(val);
+    return convertMissing != null ? convertMissing.isFillValue(val) : false;
   }
 
   @Override
   public boolean hasMissingValue() {
-    return scaleMissingUnsignedProxy.hasMissingValue();
+    return convertMissing != null ? convertMissing.hasMissingValue() : false;
   }
 
   @Override
   public double[] getMissingValues() {
-    return scaleMissingUnsignedProxy.getMissingValues();
+    return convertMissing != null ? convertMissing.getMissingValues() : new double[] {0};
   }
 
   @Override
   public boolean isMissingValue(double val) {
-    return scaleMissingUnsignedProxy.isMissingValue(val);
+    return convertMissing != null ? convertMissing.isMissingValue(val) : false;
   }
 
   /** @deprecated Use NetcdfDataset.builder() */
   @Deprecated
   @Override
   public void setFillValueIsMissing(boolean b) {
-    scaleMissingUnsignedProxy.setFillValueIsMissing(b);
+    if (convertMissing != null) {
+      convertMissing.setFillValueIsMissing(b);
+    }
   }
 
   /** @deprecated Use NetcdfDataset.builder() */
   @Deprecated
   @Override
   public void setInvalidDataIsMissing(boolean b) {
-    scaleMissingUnsignedProxy.setInvalidDataIsMissing(b);
+    if (convertMissing != null) {
+      convertMissing.setInvalidDataIsMissing(b);
+    }
+  }
+
+  public boolean missingDataIsMissing() {
+    return builder().missingDataIsMissing;
+  }
+
+  public boolean fillValueIsMissing() {
+    return builder().fillValueIsMissing;
+  }
+
+  public boolean invalidDataIsMissing() {
+    return builder().invalidDataIsMissing;
   }
 
   /** @deprecated Use NetcdfDataset.builder() */
   @Deprecated
   @Override
   public void setMissingDataIsMissing(boolean b) {
-    scaleMissingUnsignedProxy.setMissingDataIsMissing(b);
+    if (convertMissing != null) {
+      convertMissing.setMissingDataIsMissing(b);
+    }
   }
 
   @Nullable
   @Override
   public DataType getScaledOffsetType() {
-    return scaleMissingUnsignedProxy.getScaledOffsetType();
+    return scaleOffset != null ? scaleOffset.getScaledOffsetType() : dataType;
   }
 
   @Override
   public DataType getUnsignedConversionType() {
-    return scaleMissingUnsignedProxy.getUnsignedConversionType();
+    return unsignedConversion != null ? unsignedConversion.getOutType() : dataType;
   }
 
   @Override
   public DataType.Signedness getSignedness() {
-    return scaleMissingUnsignedProxy.getSignedness();
+    return unsignedConversion != null ? unsignedConversion.getSignedness() : dataType.getSignedness();
   }
 
   @Override
   public double applyScaleOffset(Number value) {
-    return scaleMissingUnsignedProxy.applyScaleOffset(value);
+    return scaleOffset != null ? scaleOffset.removeScaleOffset(value) : value.doubleValue();
   }
 
   @Override
   public Array applyScaleOffset(Array data) {
-    return scaleMissingUnsignedProxy.applyScaleOffset(data);
+    return scaleOffset != null ? scaleOffset.removeScaleOffset(data) : data;
```

Comment:
I'm not sure - that's what it was before, but that was when we only needed to be able to remove the filter for reads, now we will need to use it for writes as well. Personally I think "apply" should mean write and "remove" should mean read, but 🤷 Maybe that's too confusing because of the way this class works (which doesn't do writes)